add support for NVHPC compilers

mmiesch opened the pull request:
## Description

NVIDIA recently acquired the compilers from PGI and changed the compiler id in cmake. I find this is necessary to get this repo to compile with the latest NVIDIA/PGI compiler suite, distributed within the NVIDIA HPC SDK 21.5.

## Definition of Done

JCSDA fv3-bundle compiles with NVIDIA HPC SDK 21.5

danholdaway commented:
@mmiesch can you rebase this off the new release-stable? I moved the old one to release-stable-backup and moved in another branch as release-stable.

mmiesch commented:
> @mmiesch can you rebase this off the new release-stable? I moved the old one to release-stable-backup and moved in another branch as release-stable.

Ok - I'm not sure if I did that exactly right but the result seems to be correct.  It's rebased to the new branch.
